sdk/resourcemanager/Azure.ResourceManager.Core/docs/MigratingFromCurrentToPreview.md

```diff
@@ -0,0 +1,86 @@
+Console.WriteLine($"Subscription: {id.SubscriptionId}");
+Console.WriteLine($"ResourceGroup: {id.ResourceGroupName}");
+Console.WriteLine($"Vnet: {id.Parent.Name}");
+Console.WriteLine($"Subnet: {id.Name}");
+```
+
+### Casting to the base resource identifier
+```csharp
+string resourceId = "/subscriptions/aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee/resourceGroups/workshop2021-rg/providers/Microsoft.Network/virtualNetworks/myVnet/subnets/mySubnet";
+// Assume we don't know what type of resource id we have we can cast to the base type
+ResourceIdentifier id = resourceId;
+string property;
+if (id.TryGetSubscriptionId(out property))
+    Console.WriteLine($"Subscription: {property}");
+if (id.TryGetResourceGroupName(out property))
+    Console.WriteLine($"ResourceGroup: {property}");
+Console.WriteLine($"Vnet: {id.Parent.Name}");
+Console.WriteLine($"Subnet: {id.Name}");
+```
+
+## [Resource]Operations
+
+This represents a service client that is scoped to a particular resource.
+You can directly execute all operations on that client without needing to pass in scope
+parameters such as subscription id or resource name.
+
+*Old*
+```csharp
+    ComputeManagementClient computeClient = new ComputeManagementClient(subscriptionId, new DefaultAzureCredential());
+    string rgName = "myRgName";
+    string vmName = "myVmName";
+
+    // Each method we call needs to have the scope passed in to know which vm to operate on
+    await computeClient.StartPowerOff(rgName, vmName).WaitForCompletionAsync();
+    await computeClient.StartStart(rgName, vmName).WaitForCompletionAsync();
+```
+
+*New*
+```csharp
+    string rgName = "myRgName";
+    string vmName = "myVmName";
+    Subscription subscription = armClient.DefaultSubscription;
+    ResourceGroup resourceGroup = await subscription.GetResourceGroups().GetAsync(rgName);
+    VirtualMachine virtualMachine = await resourceGroup.GetVirtualMachines().GetAsync(vmName);
+
+    // No longer need to pass in scope parameters since the object knows the scope
+    await virtualMachine.StartPowerOff().WaitForCompletionAsync();
+    await virtualMachine.StartPowerOn().WaitForCompletionAsync();
+```
+
+This becomes more pronounced when dealing with list methods and performing operations on the response items.
+
+*Old*
+```csharp
+    ComputeManagementClient computeClient = new ComputeManagementClient(subscriptionId, new DefaultAzureCredential());
+    foreach (VirtualMachine vm in computeClient.VirtualMachines.ListAll())
+    {
+        VirtualMachineUpdate vmUpdate = new VirtualMachineUpdate();
+        vmUpdate.Tags.Add("tagKey", "tagValue");
+        // When using a subscription list, I need to parse the resource group from the resource ID in order to execute any operations
+        await computeClient.VirtualMachines.StartUpdate(GetResourceGroup(vm.Id), vm.Name, vmUpdate).WaitForCompletionAsync();
+    }
+```
+
+*New*
+```csharp
+    Subscription subscription = armClient.DefaultSubscription;
```

Comment:
This use of DefaultSubscription is not parallel.  If the user has a subscriptionId, they can use it with the new client, and this should be sown.